### ISSUE TITLE
[Request] Add guaranteed time available for finish onDismount method in frontend

### ISSUE THREAD

**Issue by steam3d** [SteamDeckHomebrew/decky-loader#612]
### Please confirm

- [X] I have searched existing issues
- [X] This issue is not a duplicate of an existing one
- [X] This is not a request for a plugin

### Feature Request Description

Some plugins must finish their background work before start them again. In cases: update, reload.

I suggest adding deferral parameter in OnDismount method that will give plugins fixed time like 1 second to stop their background before Decky will run them again.

**Comment by AAGaming00** [comment]
better option is to make onDismount async (if it isn't already?) and await it

**Comment by RodoMa92** [comment]
> better option is to make onDismount async (if it isn't already?) and await it

I agree. It seems that is not the default behaviour right now from a quick glance.

**Comment by steam3d** [comment]
The Decky simple calls:
https://github.com/SteamDeckHomebrew/decky-loader/blob/b84dcd99adbffcc2710fb916c35f698dc57fea2c/frontend/src/plugin-loader.tsx#L216
